File: android/app/src/main/java/org/haobtc/onekey/adapter/HdWalletAssetAdapter.java
package org.haobtc.onekey.adapter;

import android.content.Context;
import android.graphics.drawable.Drawable;

import androidx.annotation.Nullable;
import androidx.core.content.res.ResourcesCompat;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.google.common.base.Strings;

import org.haobtc.onekey.R;
import org.haobtc.onekey.bean.HdWalletAllAssetBean;
import org.haobtc.onekey.business.wallet.SystemConfigManager;
import org.haobtc.onekey.business.wallet.bean.WalletBalanceBean;

import java.util.List;

public class HdWalletAssetAdapter extends BaseQuickAdapter<WalletBalanceBean, BaseViewHolder> {
    private final SystemConfigManager mSystemConfigManager;

    public HdWalletAssetAdapter(Context context, @Nullable List<WalletBalanceBean> data) {
        super(R.layout.all_assets_item, data);
        mSystemConfigManager = new SystemConfigManager(context.getApplicationContext());
    }

    @Override
    protected void convert(BaseViewHolder helper, WalletBalanceBean item) {
        Drawable drawable;
        switch (item.getCoinType()) {
            default:
            case BTC:
                drawable = ResourcesCompat.getDrawable(helper.itemView.getResources(), R.drawable.token_btc, null);
                break;
            case ETH:
                drawable = ResourcesCompat.getDrawable(helper.itemView.getResources(), R.drawable.token_eth, null);
                break;
        }
        helper.setImageDrawable(R.id.imageView, drawable);
        helper.setText(R.id.text_wallet_name, item.getName()).setText(R.id.text_balance, item.getBalance());
        String strFiat = item.getBalanceFiat();
        if (!Strings.isNullOrEmpty(strFiat)) {
            helper.setText(R.id.text_fiat, "≈ " + mSystemConfigManager.getCurrentFiatSymbol() + " " + strFiat);
        }
    }
}

File: android/app/src/main/java/org/haobtc/onekey/adapter/HdWalletAssetAdapter.kt
package org.haobtc.onekey.adapter

import android.content.Context
import android.text.TextUtils
import android.widget.ImageView
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.BaseViewHolder
import com.google.common.base.Strings
import org.haobtc.onekey.R
import org.haobtc.onekey.bean.BalanceCoinInfo
import org.haobtc.onekey.bean.RemoteImage
import org.haobtc.onekey.business.assetsLogo.AssetsLogo
import org.haobtc.onekey.business.wallet.SystemConfigManager
import org.haobtc.onekey.constant.Vm
import org.haobtc.onekey.utils.CoinDisplayUtils
import java.math.BigDecimal
import java.math.RoundingMode

class HdWalletAssetAdapter(context: Context, data: List<BalanceCoinInfo?>?) : BaseQuickAdapter<BalanceCoinInfo, BaseViewHolder>(R.layout.all_assets_item, data) {
  private val mSystemConfigManager by lazy {
    SystemConfigManager(context.applicationContext)
  }

  override fun convert(helper: BaseViewHolder, item: BalanceCoinInfo) {

    if (Strings.isNullOrEmpty(item.address)) {
      val convertByCoinName = Vm.CoinType.convertByCoinName(item.coin)
      helper.getView<ImageView>(R.id.imageView)
        .setImageDrawable(mContext.getDrawable(AssetsLogo.getLogoResources(convertByCoinName)))
      helper.setText(
        R.id.text_balance,
        CoinDisplayUtils.getCoinPrecisionDisplay(item.balance, convertByCoinName)
      )
    } else {
      RemoteImage(item.icon).intoTarget(helper.getView(R.id.imageView))
      helper.setText(R.id.text_balance, getFormatBalance(item, 4))
    }
    helper.setText(R.id.text_wallet_name, item.coin.toUpperCase())

    val strFiat = if (!TextUtils.isEmpty(item.fiat) && item.fiat == "0") {
      "0.00"
    } else {
      item.fiat.substring(0, item.fiat.indexOf(" "))
    }
    if (!Strings.isNullOrEmpty(strFiat)) {
      helper.setText(R.id.text_fiat, mSystemConfigManager.currentFiatSymbol + " " + strFiat)
    }

  }

  fun getFormatBalance(item: BalanceCoinInfo, num: Int): String {
    return if (!TextUtils.isEmpty(item.balance) && !item.balance.equals("0")) {
      BigDecimal(item.balance)
        .setScale(num, RoundingMode.DOWN)
        .stripTrailingZeros().toPlainString()
    } else {
      "0"
    }
  }

}
